torch/nn/utils/rnn.py

```diff
@@ -192,7 +192,7 @@ def pack_padded_sequence(input, lengths, batch_first=False, enforce_sorted=True)
                       'values, and it will treat them as constants, likely rendering '
                       'the trace incorrect for any other combination of lengths.',
                       category=torch.jit.TracerWarning, stacklevel=2)
-    lengths = torch.as_tensor(lengths, dtype=torch.int64)
+    lengths = torch.as_tensor(lengths, dtype=torch.int64, device='cpu')
```

Comment:
is there some reason we don't make the lengths have the same device as input?

Reply:
This tensor is really only used in CPU-side loops, so we never want it to live anywhere else.